code/controllers/CronTaskController.php

```diff
@@ -97,19 +97,54 @@ public function index(SS_HTTPRequest $request) {
 	 * @param CronTask $task
 	 */
 	public function runTask(CronTask $task) {
-		$cron = Cron\CronExpression::factory($task->getSchedule());
+		$status = CronTaskStatus::get_status(get_class($task));
+
+		if ($this->isLongRunning($status)) {
+			$this->output(get_class($task).' running longer than expected. Setting status to Error');
+			SS_Log::log(get_class($task).' running longer than expected. Setting status to Error', SS_Log::WARN);
+			CronTaskStatus::update_status(get_class($task), false. 'Error');
+			return;
+		}
+
+		if ($status->Status != 'On') {
+			$this->output(get_class($task).' is in status '.$status->Status . ' and will be skipped.');
+			CronTaskStatus::update_status(get_class($task), false);
+			return;
+		}
+
+		$cron = Cron\CronExpression::factory($status->ScheduleString);
 		$isDue = $this->isTaskDue($task, $cron);
 		// Update status of this task prior to execution in case of interruption
-		CronTaskStatus::update_status(get_class($task), $isDue);
+		CronTaskStatus::update_status(get_class($task), $isDue, $isDue ? 'Running' : null);
 		if($isDue) {
 			$this->output(get_class($task).' will start now.');
 			$task->process();
+			CronTaskStatus::update_status(get_class($task), false, 'On');
 		} else {
 			$this->output(get_class($task).' will run at '.$cron->getNextRunDate()->format('Y-m-d H:i:s').'.');
 		}
 	}
 
 	/**
+	 * Check is a task running longer than defined maximum execution time
+	 *
+	 * Maximum execution time is defined in the config.yml for each task
+	 *
+	 * @param CronTaskStatus $status
+	 * @return bool
+	 */
+	private function isLongRunning($status) {
+		if ($status->Status != 'Running')
+			return false;
+
+		$minutes = Config::inst()->get($status->TaskClass, 'MaxExecutionTime');
+		if (is_null($minutes) || $minutes <= 0)
+			return false;
+
+		return strtotime(SS_Datetime::now()->getValue()) - strtotime($status->LastRun) < $minutes * 60;
```

Comment:
`SS_Datetime::now()->Format('U')` would give you the timestamp

Same with `$status->dbObject('LastRun')->Format('U')`